lib/compiler/node/index.ts

```diff
@@ -4,7 +4,27 @@ import Go from './wasm_exec.js';
 import { fileURLToPath } from 'url';
 
 export const transform: typeof types.transform = async (input, options) => {
-  return ensureServiceIsRunning().then((service) => service.transform(input, options));
+  return ensureServiceIsRunning()
+    .then((service) => service.transform(input, options))
+    .catch((err) => {
+      if (err.toString().includes('unreachable')) {
+        const title = '🐛 BUG: `@astrojs/compiler` panic';
+        const filename = options?.sourcefile ? `**root/src/${options?.sourcefile.split('/src/')[2]}**` : '';
+        const body = `### Describe the Bug
+
+\`@astrojs/compiler\` encountered an unrecoverable error when compiling the following file.
```

Comment:
Just thinking out loud: I kicked this around for a bit on the Astro end, and I think this typically comes from a syntax error. WDYT about just calling this a “syntax error?”

Reply:
well isn't the point that this isn't necessarily a syntax error, but an unrecoverable error we don't know about?

Reply:
Yeah, but I changing this to be more general (SyntaxError) and then surfacing a better error in the browser would be best

Reply:
Oh, I was considering "Unrecoverable Error" to be more general than "Syntax Error"